### PR TITLE
[T3A3][T14-A2] Tan Min Rong Samson

### PR DESCRIPTION
**Pull request by samsontmr**
Ready for review.

### CODE REVIEW

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -8,11 +8,19 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String EXAMPLE = "123, Clementi Ave 3, #12-34, 231534";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses must be in this format: a/BLOCK, STREET, UNIT, POSTAL_CODE";
+    public static final String ADDRESS_VALIDATION_REGEX = "(\\d+)(,)(.+)(,)(#| #)(\\d+)(-)(\\d+)(,)( \\d+|\\d+)";
```
Comment (nishantbudhdev) [inline src/seedu/addressbook/data/person/Address.java]: Good job on trying to use regex.

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -8,11 +8,19 @@
...
+    private static final int STREET_INDEX = 1;
+    private static final int UNIT_INDEX = 2;
+    private static final int POSTAL_CODE_INDEX = 3;
+    private final Block block;
+    private final Street street;
+    private final Unit unit;
+    private final PostalCode postalCode;
 
-    public final String value;
+    //public final String value;
```
Comment (nishantbudhdev) [inline src/seedu/addressbook/data/person/Address.java]: Remove code you don't need. Thats one of the reasons for using version control, you can rollback to earlier versions.

File: src/seedu/addressbook/data/person/Block.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    public Block(String block) {
+        this.value = block;
+        
+    }
+
+
+    /**
+     * Returns true if a given string is a valid person block.
+     */
+    public static boolean isValidBlock(String block) {
```
Comment (nishantbudhdev) [inline src/seedu/addressbook/data/person/Block.java]: Don't see the need for this, as the method is never called. The address is already validated using the `isValidAddress` function

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -25,31 +33,49 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        String[] addressArray = splitAddressIntoBlockStreetUnitPostalCode(address);
+        this.block = new Block(addressArray[BLOCK_INDEX]);
+        this.street = new Street(addressArray[STREET_INDEX]);
+        this.unit = new Unit(addressArray[UNIT_INDEX]);
+        this.postalCode = new PostalCode(addressArray[POSTAL_CODE_INDEX]);
+}
```
Comment (nishantbudhdev) [inline src/seedu/addressbook/data/person/Address.java]: Missed indentation.

**Comment by nishantbudhdev** [comment]
@samsontmr Exceptional work. You've managed to modify all the required files.

Some comments added. Please close the PR after reading comments.

File: src/seedu/addressbook/data/person/Block.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
+package seedu.addressbook.data.person;
+
+
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Block.java]: Please include a header comment to indicate what this class/method does?

**Pull request closed without merging**